Feature/files list occ command

yemkareems opened the pull request:
files:list implemented based on files:scan
List of possible usage

all user based search
occ files:list --all

path based search
occ files:list --path /admin/files

Filter by type
occ files:list --path /admin/files/Media --type video
occ files:list --path /admin/files/Media --type image
occ files:list --path /admin/files/ --type application

Filter by file size
occ files:list --path /admin/files/Media --minSize 9622690
occ files:list --path /admin/files/Media --maxSize 9622690

default sort order is desc
occ files:list --all --sort name 
to do asc
occ files:list --all --sort name --order ASC

Non recursive scan and list
occ files:list --path /admin/files/ --shallow
The above does not scan sub directories recursively

All the above params can be combined to have different result sets

yemkareems committed:
files:list added as a occ command. filter added based on type and size for files. path or --all is required to search files in the path or based on all users

yemkareems committed:
basic sorting added to list, could be improvised

yemkareems committed:
listing files and folders separately as sorting is not working as expected

yemkareems committed:
general refactoring to remove unused variables and functions, edge case in file info if no files in dir covered

AndyScherzinger commented:
Hi @yemkareems

Can you open a PR against nextcloud/server please?

That way Stephan and other would have an easier way to review the changeset.

Thanks in advance

yemkareems commented:
@AndyScherzinger Hi Andy yes sure. I will do it early tomorrow morning. Reviewing stuff once. Is it fine if i raise the PR against the nextcloud/server master branch? 

Thanks and Regards
Kareem

AndyScherzinger commented:
@yemkareems Hi Kareem,

Yes, raising it against master branch would be perfect, thank you 👍

Warm regards
Andy

yemkareems commented:
@AndyScherzinger Hi Andy I have raised a PR https://github.com/nextcloud/server/pull/43342 and tagged both you and Stephan in the comments

yemkareems committed:
Update ListFiles.php

Psalm errors fixed

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
composer run cs:fix

composer run cs:fix

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
Update ListFiles.php

exectime variable removed to fix psalm error

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
type hinting

type hinting corrected

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
PR review comments by Louis done

Scan files replaced with listFiles and all methods refactored accordingly

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
path made a argument and user extraction logic based on path changed

path made a argument and user extraction logic based on path changed

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
help doc content changed to start with all first caps

help doc content changed to start with all first caps

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
removed the user loop since only one user is there and copy right authors removed

removed the user loop since only one user is there and copy right authors removed

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
type hinting corrected

type hinting corrected

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
cs fix ran 

cs fix ran 

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
array multisort assigning to var and passing it

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
userFolder type hint to fix psalm error

userFolder type hint to fix psalm error

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
Update apps/files/lib/Command/ListFiles.php

Co-authored-by: Andy Scherzinger <info@andy-scherzinger.de>
Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
updated the description of path

updated the description of path

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
making changes to list folders as directory type

making changes to list folders as directory type

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
cs fix run for the file

cs fix run for the file

Signed-off-by: yemkareems <yemkareems@gmail.com>

yemkareems committed:
telling psalm correct path of folder

yemkareems committed:
feat: tell psalm the correct type associated with userFolder

yemkareems committed:
feat: tell psalm the correct type associated with userFolder

yemkareems committed:
feat: license wording changed as suggested
